android: add testcase templates for x15 and hikey960 builds

liuyq opened the pull request:
with the docker method used for lava jobs.
needs to specify the --testplan-path to testcases/android,
and --testplan-device-path to devices/android/,
for the submit_for_testing.py script to be run

Signed-off-by: Yongqin Liu <yongqin.liu@linaro.org>

bhcopeland commented on selftests.sh:
```diff
@@ -6,22 +6,52 @@ function run_device_selftest()
+                --test-case ${TEST} \
+                --variables variables-android.ini \
+                --device-type ${DEVICE_TYPE} \
+                --dry-run \
```
Don't need a \ at the end

liuyq replied:
removed it now to avoid confusion. I left it there because I often forget to add the backslash when I added another new parameter to the command by copying the last line.

roxell commented:
selftests.sh is just a duplication, try to make it more generic please.
start with creating a patch where you add a "linux" dir inside testcases? and maybe inside devices too if you think they are fundamentally different.
Then add this patch.
But there shouldn't be any device specific bits inside the testcase files, nor any lkft specific bits.

liuyq commented:
> selftests.sh is just a duplication, try to make it more generic please. 
reworked the update on selftests.sh, it's generic now with different parameters passed to the ./submit_for_testing.py script.

> start with creating a patch where you add a "linux" dir inside testcases? and maybe inside devices too if you think they are fundamentally different.
That will make the change big, with the path change for the existing files, then the places that call ./submit_for_testing.py needs to be updated accordingly too, but I don't know the places where ./submit_for_testing.py is called. 

If it's really necessary to change the current existing files path, we could do that after this change, just update the DIR_TESTCASES and DIR_DEVICES in the selftests.sh, and all the places where ./submit_for_testing.py is called.

liuyq commented:
> * master.jinja2

updated with some changes in the master.jinja2 file.

> project specific bits like the lkft things should be located in projects/*

could you please share me how the files under projects/lkft/ are used in the pipeline setup?
currently all the android files are used for the lkft project, there are no other user cases for now.
maybe we could re-work the templates with different projects when we need to support such user cases.
Or maybe the current files could be used directly by other cases without any change.

> don't name testcases to x15, why do you have to say arm64-v8a or armeabi-v7a? whats the specific bits?

it's only used for the x15 platform as a workaround, should not be used for other platforms.
for the  arm64-v8a or armeabi-v7a files, if you check the parameters passed, you would find the value for the abi option are different. And as we need to run them in different jobs, so we need to prepare 2 files here.(testjobs might be a better name for testcases here.)

> boot and boottime-new why? looks like they should be able to reuse things from each other... look how ltp is done.

it's not the same as LTP,  updated to use the include directive to share the common lines
they use the same tests definition but for different purpose, boot is just used to check if the device is booted, and the boottime will collect and analyze the boottime related information.

liuyq commented:
> And there are still HW names in testfilenames

dropped the hw name in the file names

> Added the test boottime in the include and not in the testcases/master/

done

> Lkft name in testcases should be in projects/lkft or projects/lkft-android or something

move templates to the projects/lkft-android directory now

> base/vts-base, maybe we have a base to much…

dropped the testcases/android/base directory now, 

> Linux dirs for linux tests are also missing.

let's do that in a separate PR, it will affect the current LTS setup and might make this PR messy.

@roxell Please help to check it again when you have time, Thanks!

roxell commented on master.jinja2:
```diff
@@ -10,7 +10,10 @@
 {% set TEST_DEFINITIONS_REPOSITORY = TEST_DEFINITIONS_REPOSITORY|default("https://github.com/Linaro/test-definitions.git") %}
 
-{% set job_timeout = lxc_deploy_timeout + lxc_boot_timeout + lxc_install_fastboot_timeout + target_deploy_timeout + TARGET_BOOT_TIMEOUT + test_timeout %}
+{% set job_timeout = job_timeout|default(lxc_deploy_timeout + lxc_boot_timeout + lxc_install_fastboot_timeout + target_deploy_timeout + TARGET_BOOT_TIMEOUT + test_timeout) %}
```
Why is this needed ?
Can't we extend this so it will work for android too instead of just overwrite it?

liuyq replied:
first, for this master.jinja2, I think it's better to not control too specific, so that others cases that don't need to think too much about the variables that it actually won't use any more. Like for the android case, the lxc_xxx_timeout variables.

liuyq replied:
anyway, I updated to not change here to avoid affects on the existing files as possible as I can.

roxell commented on master.jinja2:
```diff
@@ -20,6 +23,12 @@
 {% set enable_tests = enable_tests|default(true) %}
 
+{% set lava_device_type = lava_device_type|default("") %}
+{% set lava_job_name = lava_job_name|default("") %}
```
Not sure why these two are needed

liuyq replied:
they are needed for the L52 and L53.

roxell replied:
in all the other files its done like this:
{% block device_type %}x15{% endblock %}

I'm not convinced you need to set a variable to do this.

if you really want the variable, then you have to do the change in all the other files to and motivate why you do it.

liuyq replied:
anyway, I updated to not change here to avoid affects on the existing files as possible as I can.

roxell commented on master.jinja2:
```diff
@@ -40,17 +49,11 @@ context:
-
-{% set LAVA_JOB_VISIBILITY = LAVA_JOB_VISIBILITY|default("public") %}
-device_type: {% block device_type %}{% endblock %}
-job_name: {% block job_name %}{% endblock %}
```
look how the 'block device_type' and 'block job_name' are used, isn't that enough?
If not it should be changed everywhere.

liuyq replied:
for the current templates that defines {% block device_type %}{% endblock %}, they should still work.
I changed to use the lava_device_type or lava_job_name variables, that would make it a little easier to  specify. we only need to specify the value for the lava_device_type variable in the device template, and we even could just generate that in the ini file to specify that dynamically.

roxell replied:
fair enough, but if you want to do it, it should be done everywhere.
Also it's not that often probably never we have to change the device_type for a specific board.
Thats why its set in the devices/x15 for instance.

liuyq replied:
anyway, I updated to not change here to avoid affects on the existing files as possible as I can.

roxell commented on testcases/android/master.jinja2:
```diff
@@ -0,0 +1,107 @@
+
+{# urls #}
+{% set url_docker = url_docker|default("yongqinliu/linaro-android-docker:0.1") %}
+{% set url_test_defintions_repo = url_test_defintions_repo|default("https://github.com/Linaro/test-definitions.git") %}
```
use TEST_DEFINITIONS_REPOSITORY

liuyq replied:
will update

roxell commented on testcases/android/master.jinja2:
```diff
@@ -0,0 +1,107 @@
+{% set job_name = job_name|default("base") %}
+
+{# variables needs to be converted to be used by master.jinja2 #}
+{% set lava_job_name = "lkft-android-" + KERNEL_BRANCH + '-' + BUILD_NUMBER + '-' + job_name %}
```
'lkft-android' is project specific.

roxell commented on testcases/android/master.jinja2:
```diff
@@ -0,0 +1,107 @@
+{% block visibility %}
+  group:
+    - {{TEST_LAVA_JOB_GROUP}}
+{% endblock visibility %}
```
I would say that this is project specific too.

liuyq replied:
done

roxell commented on testcases/android/master.jinja2:
```diff
@@ -0,0 +1,107 @@
+  series: lkft
+{% block job_specific_metadata %}
+{% endblock job_specific_metadata %}
+{% endblock metadata %}
```
a lot of the metadata are also project specific.

liuyq replied:
will update, and move to the project lkft-android side

roxell commented on testcases/android/master.jinja2:
```diff
@@ -0,0 +1,107 @@
+{% block secrets %}
+secrets:
+  ARTIFACTORIAL_TOKEN: "{{ARTIFACTORIAL_TOKEN}}"
+{% endblock %}
```
project specific or?

liuyq replied:
will update

liuyq replied:
done

roxell commented on testcases/android/master.jinja2:
```diff
@@ -0,0 +1,107 @@
+{% endblock %}
+
+actions:
+{% block actions %}
```
can't 'actions' from master.jinja2 be used

liuyq replied:
Android needs its own blocks to be defined under the actions, if not defined here, how to define?

roxell replied:
can you share a job definition file from LAVA so I can understand where this have to be added a second time?

liuyq replied:
@roxell you mean the L62  or the lines after L63? L62 is not necessary, i will remove it. in the generated lava job definition, there is only one "actions:". For L63, it defined the blocks that android jobs need to overwrite, like here: https://lkft.validation.linaro.org/scheduler/job/4772849/definition

liuyq replied:
in the new update, removed the "actions:" line here.

roxell replied:
you should be use '{{ super() }}' inside the '{% block actions %}'

liuyq replied:
the actions defined here are common actions, and concept of the actions is too big.

roxell commented on testcases/android/master.jinja2:
```diff
@@ -0,0 +1,107 @@
+{% endblock test_common_actions %}
+
+{% block test_target %}
+{% include "include/android-test-target.jinja2" %}
```
this file and include/test_target.jinja2 isn't that different.

liuyq replied:
but don't you think include/test_target.jinja2 is a little over complex to be read and to be used?

roxell replied:
no.

liuyq replied:
updated to change  include/test_target.jinja2 to support the docker method now.

liuyq commented:
@roxell updated again, please help to check it when you have time, thanks!

roxell commented on include/boot_target/android-boot.jinja2:
```diff
@@ -0,0 +1,9 @@
+    - 'root@(.*):/#'
+    - 'console:/'
+    timeout:
+      minutes: {{timeout_boot}}
```
we have a variable called TARGET_BOOT_TIMEOUT for this

liuyq replied:
will update to use TARGET_BOOT_TIMEOUT

roxell commented on projects/lkft-android/master.jinja2:
```diff
@@ -0,0 +1,69 @@
+  toolchain: "{{TEST_METADATA_TOOLCHAIN}}"
+  series: lkft
+{% block job_specific_metadata %}
+{% endblock job_specific_metadata %}
```
This can be dropped. Please use  '{{ super() }}'

liuyq replied:
yes, it could be, but instead of 
```
{% block metadata %}
{{ super() }}
...
{% endblock %}
```
we only need to write
```
{% block job_specific_metadata %}
...
{% endblock job_specific_metadata %}
```
so that we don't need to remember to write the '{{ super() }}' line, and job_specific_metadata block makes it more clear on what the lines in it are.

roxell replied:
I guess you understand but I mean like this: http://ix.io/3U06

I think its clear enough since you specify metadata in a test specific file...

liuyq replied:
here is my reason to use job_specific_metadata `so that we don't need to remember to write the '{{ super() }}' line, and job_specific_metadata block makes it more clear on what the lines in it are.` 
what's advantage there with the "{{super}}" method?

roxell commented on projects/lkft-android/master.jinja2:
```diff
@@ -0,0 +1,69 @@
+  series: lkft
+{% block job_specific_metadata %}
+{% endblock job_specific_metadata %}
+{% endblock metadata %}
```
maybe it would make sense to do like the other projects and create a file in projects/<project>/include/metadata.jinja2 ?

liuyq replied:
I thought of it, but as it's only used here, so no meaning to maintain 2 files.
If the same lines are used by another place in the future, then we'd better to create a separate metadata file, and include it for different places.

roxell replied:
So if I want to do some android testing with this repository, where should I place my project specific file?

liuyq replied:
in the project specific directory. if it could share some of the lines with other projects, say the metadata lines with the lkft-android project, then create the include file under the include directory, and use the include directive for both projects.

roxell replied:
fair enough.

roxell commented on testcases/android/cts.yaml:
```diff
@@ -0,0 +1,23 @@
+
+{% set cts_test_params = cts_test_params | default("cts") %}
+
+{% block job_specific_metadata %}
```
You should be able to say:
{% block metadata %}
{{ super() }}
cts-url: "{{TEST_CTS_URL}}/android-cts.zip"
...
{% endblock metadata %}

liuyq replied:
job_specific_metadata will make it more clear it's only for this job.

roxell commented on testcases/android/master.jinja2:
```diff
@@ -0,0 +1,51 @@
+{% set timeout_between_download_deploy = timeout_between_download_deploy|default(0) %}
+{% set timeout_deploy = timeout_deploy|default(25) %}
+{% set timeout_between_deploy_boot = timeout_between_deploy_boot|default(0) %}
+{% set timeout_boot = TARGET_BOOT_TIMEOUT|default(10) %}
```
Whats the rational behind inventing your own variables?

liuyq replied:
will update to use TARGET_BOOT_TIMEOUT

roxell reviewed (commented):
a general comment, currently in the repository there is 3 way of booting, fastboot, nfs and qemu, plus we can do that in docker or lxc...
and what I can see android can do a bit more with two A and B, which is fastboot + a recovery image.
So maybe just add that or?

also the ordering how things are include doesn't follow the current structure, I see a benefit of trying to keep it as similar as possible what do you think?

liuyq commented on devices/android/device-base:
```diff
@@ -0,0 +1,47 @@
+      {% block partitions_names %}
+      {% for partition in partitions_names %}
+      {{partition}}:
+        url: downloads://{{partitions_names[partition]}}
```
no, the url might have the suffix like xz, it's not always the same.

roxell replied:
what does the suffix have to do with it?
$ basename https:/path/to/some/image.img.xz
image.img.xz
$ basename https:/path/to/some/image.img   
image.img

liuyq replied:
say if the url is https:/path/to/some/image.img.xz, the basename would be image.img.xz, "url: downloads://image.img.xz" could not be used for the deploy action here. that's why we need to the partitions_names and partitions_urls. And if there are postprocess: for the download action, then there might be more difference there.

roxell replied:
I'm more for adding that if its needed if there are more difference.

liuyq replied:
sorry, not get your point here, could you please explain it with some examples?

liuyq commented on devices/android/hikey960:
```diff
@@ -0,0 +1,79 @@
+                                                    "system": f_name_system,
+                                                    "vendor": f_name_vendor
+                                                   }) %}
+{% endif %}
```
will update to define the common variables with ptable, boot, userdata defined.

liuyq replied:
updated

liuyq commented on devices/android/x15:
```diff
@@ -0,0 +1,47 @@
+{% extends "device-base" %}
+
+{% set TAGS = TAGS|default(["hdmi-dongle"]) %}
+{% set lava_device_type = lava_device_type|default("x15") %}
```
then I don't need write "{% block device_type %}x15{% endblock %}" here.
first x15 is a hard coded string second, {% block device_type %} is kind of a common thing, the leave it in the common file.

roxell replied:
I'm sorry, if you want to change that you have to change that in all the files in this repository.

liuyq replied:
you win, I will update to use the hard coded method.

liuyq replied:
done

liuyq commented on include/boot_target/android-boot.jinja2:
```diff
@@ -0,0 +1,9 @@
+- boot:
+    docker:
+      image: {{url_docker}}
```
will update to use DOCKER_IMAGE

liuyq replied:
updated

liuyq commented:
> a general comment, currently in the repository there is 3 way of booting, fastboot, nfs and qemu, plus we can do that in docker or lxc...
> and what I can see android can do a bit more with two A and B, which is fastboot + a recovery image.
> So maybe just add that or?

where you want to add that? I'd like to have the android parts in a separate file to make it easy to read.
there are too many if checks in the include/fastboot.jinja2, I really don't want to read any line in that file.:(

> also the ordering how things are include doesn't follow the current structure, I see a benefit of trying to keep it as similar as possible what do you think?

The includes here are really complex, honestly I still do not quite understand the include order/rules for the LTS setup.

btw, include files should be only included, but there are include files to be extended....

liuyq commented:
@roxell updated to not define lava_device_type now, and replied your comments, please help check it again.

roxell commented on devices/android/device-base:
```diff
@@ -0,0 +1,47 @@
+
+{% set partitions_urls = partitions_urls|default({}) %}
+{% set partitions_names = partitions_names|default({}) %}
+{% set timeout_download = timeout_download |default(10) %}
```
following the other timeouts we have they are in the format '*_timeout'

liuyq replied:
updated

roxell commented on devices/android/device-base:
```diff
@@ -0,0 +1,47 @@
+{% if partitions_names %}
+- deploy:
+    timeout:
+      minutes: {{timeout_deploy}}
```
target_deploy_timeout is what is used here elsewhere in the project

roxell commented on devices/android/device-base:
```diff
@@ -0,0 +1,47 @@
+      {% endfor %}
+      {% endblock partitions_urls %}
+{% endif %}
+{% endblock download_target %}
```
how much of this can be pulled in from the fastboot file?

liuyq replied:
you mean the root fastboot.jinja2 file? if it's one include file, then we could think about using it...
maybe we could do some changes in the future on the deploy_target to make it more generic that can be used everywhere, but it's not something at this time.

roxell replied:
ok

roxell commented on projects/lkft-android/master.jinja2:
```diff
@@ -0,0 +1,68 @@
+
+{% set timeout_common_tests_after_boot = timeout_common_tests_after_boot|default(20) %}
+
+{% set lava_job_name = "lkft-android-" + KERNEL_BRANCH + '-' + BUILD_NUMBER + '-' + job_name %}
```
Looking at the default job_name
{% block job_name %}{{PROJECT_NAME}}-{{OS_INFO}}-{{test_name}}-{{KERNEL_BRANCH}}-{{BUILD_NUMBER}}{% endblock job_name %}

wouldn't it be sufficient to use this?
What do  you miss here?

liuyq replied:
first I see it's defined in the testcases/master/template-master.jinja2 file, why it could not be defined in the root master.jinja2 file?

and another question, with the format of "{{PROJECT_NAME}}-{{OS_INFO}}-{{test_name}}-{{KERNEL_BRANCH}}-{{BUILD_NUMBER}}", when we need to search all the jobs for one build(BUILD_NUMBER is known), what should be input in the lava textbox? 

when OS_INFO is not defined, there will be "--" in the job name.

it's to override the job_name block, why use the {{test_name}} variable? test_name should be the name of test under the test actions, and there might be multiple tests defined in on job with different test_name.

roxell replied:
OS_INFO: thats correct, OS_INFO for lkft should be OE or Debian, for android I expect it to be android or should it be something else?

Format can be changed, we just have to come to an agreement what format make sense.

I have to look why its in template-master.jinja2 and not in the root master.jinja2

liuyq replied:
OK, when it's defined in the root master.jinja2 file, and in the format of "{{PROJECT_NAME}}-{{OS_INFO}}-{{KERNEL_BRANCH}}-{{BUILD_NUMBER}}-{{job_name}}", the android definition here could be removed with the use of that in the root master.jinja2. I think that could be done in a separate change. Moved the {{job_name}} to the last place, that would help us to search all the jobs of the same build. like when we want to check all the jobs of the build#2532 of android13-5.10 branch db845c builds, we could input lkft-android-android13-5.10-2532 in the search box here:
https://lkft.validation.linaro.org/scheduler/device_type/dragonboard-845c?dt_length=25&dt_search=lkft-android-android13-5.10-2532&dt_page=1#dt_

liuyq replied:
created a issue here: https://github.com/Linaro/lava-test-plans/issues/182

roxell commented on projects/lkft-android/master.jinja2:
```diff
@@ -0,0 +1,68 @@
+{% extends "testcases/android/master.jinja2" %}
```
Can I generate an android job file without any project set?

liuyq replied:
yes, the default selftests.sh check will have no project specified.

roxell commented on projects/lkft-android/vts-kernel-armeabi-v7a.yaml:
```diff
@@ -0,0 +1,7 @@
+{% extends "testcases/android/vts.yaml" %}
+
+{% set test_timeout = 480 %}
+{% set job_name = "vts-kernel-armeabi-v7a" %}
```
this should probably be test_name

liuyq replied:
no, it's the job name, the test_name is hard coded in the test definitions part. like for the vts job, the test names is vts-test, which will be the test_suite name in lava result, like here: https://lkft.validation.linaro.org/results/4820252

roxell replied:
that should be changed.

liuyq replied:
If you meant the test_name should be changed to job_name, then let's do that in a separate change.
If you meant the job_name here should be changed to test_name, then please give more explanation.

roxell reviewed (commented):
hikey960 and hi960-hikey are the same device, why do we create hikey960? also in that file there is a lot of fastboot specific bits inside same with x15.

liuyq commented:
> hikey960 and hi960-hikey are the same device, why do we create hikey960?

hikey960 is used by all the android build configs currently, the device file here does not need to be the same as the device_type name defined in the lava.

btw, why it's not defined in the root master.jinja2 like this:
`device_type: {{device_type}}`

then you won't need to define like `{% block device_type %}dragonboard-410c{% endblock %}` everywhere?

 > also in that file there is a lot of fastboot specific bits inside same with x15.

Sorry, please give me the examples on why you think they are the same.
if you mean the definitions on super and userdata, I think that's OK, each device file defines it's own files that supported for different android versions, without affecting on other devices.
defining them in the device-base would just increase the maintenance work to look after the definitions, to check if the variable name is the same, etc.

roxell commented:
> > hikey960 and hi960-hikey are the same device, why do we create hikey960?
> 
> hikey960 is used by all the android build configs currently, the device file here does not need to be the same as the device_type name defined in the lava.

if we play with the idea that the files in the devices directory are only different device-types in LAVA, even though right now the file name and device-type in lava doesn't match.
This files should only have the device specific config to boot a kernel on the device.
For now we only use fastboot for x15, hikey and dragonboards, so these files should extend the fastboot configs.

If we now do different things booting linux/android on x15 or hikey devices we have boot 1 and boot 2 on android and linux only use one boot url/file... however, this should be part of the fastboot.jinja2 file. and if the fastboot file gets too complicated with boot1 and boot 2 url/files in it ?
If thats the case what do we do then do you think?

> 
> btw, why it's not defined in the root master.jinja2 like this: `device_type: {{device_type}}`
> 
> then you won't need to define like `{% block device_type %}dragonboard-410c{% endblock %}` everywhere?
> 
> created a issue here: #183

That may be a way forward yes, but that is not needed to do before this commit.

> 
> > also in that file there is a lot of fastboot specific bits inside same with x15.
> 
> Sorry, please give me the examples on why you think they are the same. if you mean the definitions on super and userdata, I think that's OK, each device file defines it's own files that supported for different android versions, without affecting on other devices. defining them in the device-base would just increase the maintenance work to look after the definitions, to check if the variable name is the same, etc.

That should be solved with a URL/FILENAME like ROOTFS_URL or KERNEL_URL or what am I missing?

roxell commented:
For now I think we should just prefix the android tests instead of creating a android directory in testcases.

liuyq commented on projects/lkft-android/fastboot.jinja2:
```diff
@@ -0,0 +1,119 @@
+{% set pre_boot_timeout = pre_boot_timeout|default(0) %}
+{% set post_boot_timeout = post_boot_timeout|default(10) %}
+
+{# target_deploy_timeout is used to calculate the job_timeout defined in the root master.jinja2 file #}
```
Just want to mention here, the following lines on the target_deploy_timeout will be deleted after the PR here merged: https://github.com/Linaro/lava-test-plans/pull/187/files

roxell replied:
awesome

roxell commented:
Can we break the patch in smaller patches.

roxell commented on devices/x15:
```diff
@@ -4,7 +4,7 @@
 {% set ROOTFS_URL_COMP = ROOTFS_URL_COMP|default("gz") %}
 
-{% set boot_method = "u-boot" %}
+{% set boot_method = boot_method|default("u-boot") %}
```
one patch for this.

liuyq replied:
done

roxell commented on include/android-boottime-definition.jinja2:
```diff
@@ -0,0 +1,10 @@
+          params:
+            ANDROID_VERSION: master
+            OPERATION: {{ test_operation }}
+            COLLECT_NO: "{{ test_index }}"
```
are the quotes needed here?

roxell commented on include/android-boottime-definition.jinja2:
```diff
@@ -0,0 +1,10 @@
+        - repository: {{ TEST_DEFINITIONS_REPOSITORY }}
```
in the changelog explain why this deserves its own file. That it will be used by more tests, that will be added.

liuyq commented:
@roxell updated to split it into several smaller changes, please help check it again

roxell commented on projects/lkft-android/include/lkft-common-actions.jinja2:
```diff
@@ -0,0 +1,15 @@
+            - lava-test-case "android-boot-lsusb-v" --shell lsusb -v
+            - lava-test-case "android-boot-boot-completed" --shell "while ! adb shell getprop sys.boot_completed|grep 1; do sleep 2; done"
+            - lava-test-case "android-boot-set-power-stayon" --shell adb shell su 0 svc power stayon true
+            - lava-test-case "android-boot-screencap" --shell adb shell screencap -p /data/local/tmp/screencap.png
```
What makes this file specific to lkft-android? and not just android?

liuyq replied:
this is just some commands that run after the boot, other cases might not  need to run or need to run all of them, so it's better to have these commands for project only instead of putting them in the common place.

roxell commented on projects/lkft-android/variables-android.ini:
```diff
@@ -0,0 +1,30 @@
+TEST_VTS_URL=test-vts-url-unknow
+# vairiable might be defined in the build config for not
+# if not defined, need to give the default value
+TEST_LAVA_JOB_PRIORITY=test-lava-job-priority-unknown
```
drop prefix TEST_

roxell commented on testcases/master/template-android.jinja2:
```diff
@@ -0,0 +1,17 @@
+{% set use_docker_image_for_test_target = use_docker_image_for_test_target|default(true) %}
+
+{# variables needs to be converted to be used by the root master.jinja2 #}
+{% set LAVA_JOB_PRIORITY = LAVA_JOB_PRIORITY|default("medium") %}
```
Should be enough if this is set in the variables.(ini|yaml).

liuyq replied:
it could be set in the ini/yaml file, but it's not obvious to read that we set the default medium for LAVA_JOB_PRIORITY.

roxell commented on testcases/master/template-android.jinja2:
```diff
@@ -0,0 +1,17 @@
+
+{# urls #}
+{% set DOCKER_IMAGE = DOCKER_IMAGE|default("yongqinliu/linaro-android-docker:0.1") %}
+{% set use_docker_image_for_test_target = use_docker_image_for_test_target|default(true) %}
```
Maybe there two variables belong in projects/lkft-android/fastboot.jinja2 ?

liuyq replied:
no, I put them here just want to make it clear that all the android jobs should be docker method, at least by default.

roxell commented on testcases/master/template-android.jinja2:
```diff
@@ -0,0 +1,17 @@
+{########## LAVA JOB DEFINITION ##############}
+{% block test_target %}
+    {{ super() }}
+    {% block test_target_definitions %}{% endblock test_target_definitions %}
```
Not sure why we have both test_target and test_target_definitions.

liuyq replied:
with test_target_definitions, we don't need to write {{super}} everywhere.
instead of 
```yaml
{% block test_target %}
    {{ super() }}
    ...
{% endblock test_target %}
```
we only need 
```yaml
{% block test_target_definitions %}
    ...
{% endblock test_target_definitions %}
```

roxell commented on testcases/master/template-android.jinja2:
```diff
@@ -0,0 +1,17 @@
+{% extends "testcases/master/template-master.jinja2" %}
+
+{% set OS_INFO = OS_INFO|default("android") %}
+{% set DEPLOY_OS = DEPLOY_OS|default("android") %}
```
Maybe these two variables should be defined somewhere else and not in the test?

liuyq replied:
I am not sure if there is any other better places, but this file is the root file that all android jobs should extend, so I defined them here.

liuyq commented:
@roxell let's split this PR to several PRs so that we could have some progress there,
Here is one PR for the first commit, let's have it merged first: https://github.com/Linaro/lava-test-plans/pull/191
